src/status_im/contexts/centralized_metrics/events.cljs

```diff
@@ -45,3 +45,7 @@
                           #(rf/dispatch [:profile.login/login-with-biometric-if-available
                                          (get-in db [:profile/login :key-uid])]))
               :shell?   true}]]]})))
+
+(rf/reg-event-fx :centralized-metrics/track-event
+ "Track an event, :centralized-metrics/track-event event-name {:kebab_case_key value}"
+ (fn [_ _] {}))
```

Comment:
We can explain why this workaround exist with an empty function body, mentioning the global interceptor will process `:centralized-metrics/track-event`.

Reply:
@mohsen-ghafouri @ilmotta maybe would be better to do this without the interceptor at all? We could extract the native call from the interceptor and use it in the event (through an fx), like here:

```clj
;; Extracted from `centralized-metrics-interceptor`
(defn add-metric
  [event]
  (log/debug "tracking event" event)
  (native-module/add-centralized-metric event))

(defn centralized-metrics-interceptor
  [context]
  (let [event     (tracking/tracked-event (interceptor/get-coeffect context :event))
        can-push? (-> context (interceptor/get-coeffect :event) push-event?)]
    (when (and event can-push?)
      (add-metric event)))
  context)

(rf/reg-fx :effects.centralized-metrics/track-event
 (fn [event]
   (add-metric event)))

(rf/reg-event-fx
 :centralized-metrics/track-event
 (fn [{:keys [db]} [event-id data]]
   (when (push-event? db)
     {:fx [[:effect.metrics/track
            (tracking/key-value-event event-id data)]]})))
```

Reply:
I agree with this approach as also i suggested to call directly a func https://github.com/status-im/status-mobile/pull/21361#issuecomment-2393776795, but Icaro mentioned it will better to use event. so I belive your suggestion make sense and better to extract it from interceptor.

Reply:
This is the correct approach. This is what I had suggested @mohsen-ghafouri in my reply to @clauxx and was my first comment to you https://github.com/status-im/status-mobile/pull/21361#issuecomment-2393467442 (`The solution 2 with an empty event :centralized-metrics/track-event relying on the interceptor feels quite hacky.`)

> Dispatching a separate event as you suggest can work well, I agree 👍🏼 What should the handler body look like? I'm imagining the body would use a new re-frame effect to execute the effectful call to native-module/add-centralized-metric (basically what we do in the global interceptor).
>
> -- https://github.com/status-im/status-mobile/pull/21280#discussion_r1784333102